### PR TITLE
Converter refactor

### PR DESCRIPTION
**Pull request by chaitanyaavi**

### CODE REVIEW

File: python/avi/f5_converter/conversion_util.py [hunk 1]
```diff
@@ -29,15 +29,52 @@ def upload_file(file_path):
     return file_str
 
 
-def update_skipped_attributes(skipped, indirect_list, ignore_dict, object):
+def update_skipped_attributes(skipped, indirect_list, ignore_dict,
+                              f5_object, user_ignore=None):
```
Comment (grastogi23) [inline python/avi/f5_converter/conversion_util.py]: need function doc strings

File: python/avi/f5_converter/conversion_util.py [hunk 2]
```diff
@@ -87,65 +124,84 @@ def get_cc_algo_val(cc_algo):
...
+    :param f5_id: Name oconv_f object
+    :param conv_status: dict of conversion status
+    :param avi_object: Converted objectconverted avi object
+    """
+    global csv_writer
+    row = {
+        'F5 type': f5_type,
+        'F5 SubType': f5_sub_type,
+        'F5 ID': f5_id,
+        'Status': conv_status.get('status'),
```
Comment (grastogi23) [inline python/avi/f5_converter/conversion_util.py]: might be good to have a check or have a default value for the conv_status. there should not be exception raised here.

File: python/avi/f5_converter/f5_config_converter.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        LOG.debug("Converted persistence profiles")
+
+        vs_conv = VSConfigConv.get_instance(version)
+        vs_conv.convert(f5_config, avi_config_dict, vs_state, user_ignore)
+        LOG.debug("Converted VS")
+
+        conv_utils.cleanup_config(avi_config_dict)
+
+    except:
+        LOG.error("Conversion error", exc_info=True)
```
Comment (grastogi23) [inline python/avi/f5_converter/f5_config_converter.py:53]: we should print the stack trace and other debug messages to help understand what happend ere.

File: python/avi/f5_converter/f5_config_converter.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    csv_file = open(status_file, 'w')
+    conv_utils.add_csv_headers(csv_file)
+    avi_config_dict = {}
+    try:
+        mon_conv = MonitorConfigConv.get_instance(version)
+        mon_conv.convert(f5_config, avi_config_dict, input_dir, user_ignore)
+
+        pool_conv = PoolConfigConv.get_instance(version)
+        pool_conv.convert(f5_config, avi_config_dict, user_ignore)
+        LOG.debug("Converted pools")
```
Comment (grastogi23) [inline python/avi/f5_converter/f5_config_converter.py]: can we print the number of pools, number of objs etc. as the conversion happens

File: python/avi/f5_converter/test/certs/sample_monitor [hunk 5]
```diff
@@ -0,0 +1,60 @@
+#!/bin/sh
```
Comment (grastogi23) [inline python/avi/f5_converter/test/certs/sample_monitor]: remove this from the code base.

File: python/avi/f5_converter/f5_config_converter.py (hunk 3 above)
Comment (chaitanyaavi) [inline python/avi/f5_converter/f5_config_converter.py:53]: exc_info=True prints the stack trace

**Pull request merged**